src/framework/components/element/text-element.js

```diff
@@ -814,6 +856,31 @@ Object.assign(pc, function () {
                     meshInfo.uvs[quad * 4 * 2 + 6] = uv[0];
                     meshInfo.uvs[quad * 4 * 2 + 7] = uv[3];
 
+                    // set per-vertex color
+                    if (this._symbolColors) {
+                        color = this._colorPalette.slice(this._symbolColors[i] * 3, this._symbolColors[i] * 3 + 3);
```

Comment:
Do we have to have `color` be an array? If we could avoid that we can skip that allocation (including the allocation in the beginning that assigns color to white)

Reply:
made them separate members.